ride_share.rb

```diff
@@ -0,0 +1,90 @@
+driver_data_for_use = [
+    {DR0004: {date: "3rd Feb 2016", cost: 5, rider_id: "RD0022", rating: 5}},
+    {DR0001: {date: "3rd Feb 2016", cost: 10, rider_id: "RD0003", rating: 3}},
+    {DR0002: {date: "3rd Feb 2016", cost: 25, rider_id: "RD0073", rating: 5}},
+    {DR0001: {date: "3rd Feb 2016", cost: 30, rider_id: "RD0015", rating: 4}},
+    {DR0003: {date: "4th Feb 2016", cost: 5, rider_id: "RD0066", rating: 5}},
+    {DR0004: {date: "4th Feb 2016", cost: 10, rider_id: "RD0022", rating: 4}},
+    {DR0002: {date: "4th Feb 2016", cost: 15, rider_id: "RD0013", rating: 1}},
+    {DR0003: {date: "5th Feb 2016", cost: 50, rider_id: "RD0003", rating: 2}},
+    {DR0002: {date: "5th Feb 2016", cost: 35, rider_id: "RD0066", rating: 3}},
+    {DR0004: {date: "5th Feb 2016", cost: 20, rider_id: "RD0073", rating: 5}},
+    {DR0001: {date: "5th Feb 2016", cost: 45, rider_id: "RD0003", rating: 2}}
+]
+
+# 1 --- the number of rides each driver has given 
+driver_and_trips = {}
+
+driver_data_for_use.each do |trip|
+    trip.each do |key, value|
+        if driver_and_trips.include?(key) == false
+            driver_and_trips[key] = 1
+        else
+            driver_and_trips[key] += 1
+        end 
+    end
+end 
+
+driver_and_trips.each do |driver, trip|
+    puts "#{driver} gave a total of #{trip} rides"
+end 
+driver_and_trips
```

Comment:
This line doesn't do anything! The computer just goes "What's the value of `driver_and_trips` and then moves on."